File: scripts/deploy.ts
import { ethers } from "hardhat";

async function main() {
  const currentTimestampInSeconds = Math.round(Date.now() / 1000);
  const oneYearInSeconds = 365n * 24n * 60n * 60n;
  const unlockTime = currentTimestampInSeconds + Number(oneYearInSeconds);

  const lockedAmount = ethers.utils.parseEther("0.00000001");

  try {
    const Lock = await ethers.getContractFactory("Lock");
    const lock = await Lock.deploy(unlockTime, { value: lockedAmount });
    
    await lock.deployed();

    console.log(`Lock with 0.00000001 ETH and unlock timestamp ${unlockTime} deployed to ${lock.address}`);
    console.log(`Block explorer URL: https://blockscout.scroll.io/address/${lock.address}`);
  } catch (error) {
    console.error("Error deploying contract:", error);
    process.exitCode = 1;
  }
}

// Use async/await pattern with error handling
main().catch((error) => {
  console.error(error);
  process.exitCode = 1;
});

File: scripts/deploy.py
from brownie import accounts, Lock, Wei
import time

BROWNIE_ACCOUNT_NAME = 'main-account'
ONE_YEAR_IN_SECS = 365 * 24 * 60 * 60
TOTAL_LOCKED_AMOUNT_IN_ETHER = '0.00000001'
    
def deploy():

    account = accounts.load(BROWNIE_ACCOUNT_NAME)

    currentTimestampInSeconds = round(time.time())
    unlockTime = currentTimestampInSeconds + ONE_YEAR_IN_SECS
    lockedAmount = Wei(f'{TOTAL_LOCKED_AMOUNT_IN_ETHER} ether')
    
    lock = Lock.deploy(unlockTime, {"from": account, "value": lockedAmount})

    print(f'Lock with {TOTAL_LOCKED_AMOUNT_IN_ETHER} ETH and unlock timestamp {unlockTime} deployed to {lock.address}')

    # Pre-alpha block explorer.
    # print(f'Block explorer URL: https://l2scan.scroll.io/address/{lock.address}')
    # Testnet block explorer.
    # print(f'Block explorer URL: https://sepolia.scrollscan.com/address/{lock.address}')
    # Mainnet block explorer.
    print(f'Block explorer URL: https://blockscout.scroll.io/address/{lock.address}')
    
def main():

    deploy()